integration-libs/opf/base/core/services/opf-payment-hosted-fields.service.ts

```diff
@@ -0,0 +1,234 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { ActiveCartFacade } from '@spartacus/cart/base/root';
+import {
+  backOff,
+  GlobalMessageService,
+  RoutingService,
+  UserIdService,
+  WindowRef,
+} from '@spartacus/core';
+import { Order } from '@spartacus/order/root';
+
+import { combineLatest, EMPTY, from, Observable, throwError } from 'rxjs';
+import {
+  catchError,
+  concatMap,
+  filter,
+  map,
+  switchMap,
+  take,
+  tap,
+} from 'rxjs/operators';
+
+import {
+  defaultError,
+  MerchantCallback,
+  OpfOrderFacade,
+  OpfOtpFacade,
+  OpfPaymentError,
+  PaymentErrorType,
+  PaymentMethod,
+  SubmitCompleteInput,
+  SubmitCompleteRequest,
+  SubmitCompleteResponse,
+  SubmitInput,
+  SubmitRequest,
+  SubmitResponse,
+  SubmitStatus,
+} from '@spartacus/opf/base/root';
+import { OpfPaymentConnector } from '../connectors/opf-payment.connector';
+import { OpfPaymentErrorHandlerService } from '../services/opf-payment-error-handler.service';
+import {
+  isAuthorizationError,
+  opfAuthorizationErrorRetry,
+} from '../utils/opf-occ-http-error-handlers';
+import { getBrowserInfo } from '../utils/opf-payment-utils';
+
+@Injectable()
+export class OpfPaymentHostedFieldsService {
+  constructor(
+    protected opfPaymentConnector: OpfPaymentConnector,
+    protected winRef: WindowRef,
+    protected opfOtpFacade: OpfOtpFacade,
+    protected activeCartFacade: ActiveCartFacade,
+    protected userIdService: UserIdService,
+    protected routingService: RoutingService,
+    protected opfOrderFacade: OpfOrderFacade,
+    protected globalMessageService: GlobalMessageService,
+    protected opfPaymentErrorHandlerService: OpfPaymentErrorHandlerService
+  ) {}
+
+  submitPayment(submitInput: SubmitInput): Observable<boolean> {
```

Comment:
[MEDIUM]

In general, let's avoid using booleans and instead use meaningful enums. The reason for this is that enums are more adaptable to future changes, such as adding a new states (third, fourth) whenever needed. 

Note 1: Please review the entire PR for similar cases and use enums instead of booleans. 

Note 2. Wherever it's _truly justified_ to return `true`/`false` (i.e. when we don't return a status, but answer a "question"), then I suggest that the method name should start with something like a `is...` or `should...` to *sound like a question* that can be answered natually with a `true` or `false` value (e.g. `isVisible()`) etc.